Framework/Algorithms/src/MultipleScatteringCorrection.cpp

```diff
@@ -236,111 +515,238 @@ void MultipleScatteringCorrection::exec() {
         wvBinsIndex = specSize - m_xStep - 1;
       }
     }
-
     // Interpolate linearly between points separated by m_xStep,
     // last point required
     if (m_xStep > 1) {
-      auto histNew = m_outputWS->histogram(workspaceIndex);
+      auto histNew = outws->histogram(workspaceIndex);
       interpolateLinearInplace(histNew, m_xStep);
-      m_outputWS->setHistogram(workspaceIndex, histNew);
+      outws->setHistogram(workspaceIndex, histNew);
     }
-
     prog.report();
-
     PARALLEL_END_INTERUPT_REGION
   }
   PARALLEL_CHECK_INTERUPT_REGION
-
   g_log.notice() << "finished integration.\n";
+}
 
-  // set the output workspace group
-  // TODO: additional workspace will be added in as we are gradually implementing
-  //       support for container-sample scattering correction
-  const std::string outWSName = getProperty("OutputWorkspace");
-  std::vector<std::string> names;
-  names.emplace_back(outWSName + "_sampleOnly");
-  API::AnalysisDataService::Instance().addOrReplace(names.back(), m_outputWS);
-  // group
-  auto group = createChildAlgorithm("GroupWorkspaces");
-  group->initialize();
-  group->setProperty("InputWorkspaces", names);
-  group->setProperty("OutputWorkspace", outWSName);
-  group->execute();
-  API::WorkspaceGroup_sptr outWS = group->getProperty("OutputWorkspace");
-
-  setProperty("OutputWorkspace", outWS);
+/**
+ * @brief compute LS1s within given shape for single component case
+ *
+ * @param distGraber
+ * @param LS1s
+ * @param shape
+ */
+void MultipleScatteringCorrection::calculateLS1s(const MultipleScatteringCorrectionDistGraber &distGraber, //
+                                                 std::vector<double> &LS1s,                                //
+                                                 const Geometry::IObject &shape) const {
+  const auto sourcePos = m_inputWS->getInstrument()->getSource()->getPos();
+  const int64_t numVolumeElements = distGraber.m_numVolumeElements;
+  Track trackerLS1(V3D{0, 0, 1}, V3D{0, 0, 1});
+
+  for (int64_t idx = 0; idx < numVolumeElements; ++idx) {
+    const auto pos = distGraber.m_elementPositions[idx];
+    const auto vec = getDirection(sourcePos, pos);
+    //
+    trackerLS1.reset(sourcePos, vec);
+    trackerLS1.clearIntersectionResults();
+    const auto dist1 = getDistanceInsideObject(shape, trackerLS1);
+    trackerLS1.reset(pos, vec);
+    trackerLS1.clearIntersectionResults();
+    const auto dist2 = getDistanceInsideObject(shape, trackerLS1);
+    //
+    LS1s[idx] = (dist1 - dist2);
+  }
 }
 
 /**
- * @brief parse and assign corresponding values from input properties
+ * @brief compute LS1s within given shape for sample and container case
  *
+ * @param distGraberContainer
```

Comment:
Remove empty doxygen

Reply:
I added the missing docs for each param instead